fix new link in installation_guide.md

Gutopro opened the pull request:
A new link for an installation  guide on bare metal operators which can be found here https://github.com/metal3-io/baremetal-operator/blob/main/docs/dev-setup.md

tuminoid commented:
/ok-to-test

Gutopro commented:
hello @tuminoid I'm in a bit of confusion, I made an initial commit on the branch which I requested a pull from, after some time I fixed another link on the same branch and made a commit on it, now my confusion arose from my inability to create a new pull request, will both commits be tested and treated as a single pull request since they affect the same file and are on the same branch or is there something wrong I did or should do correctly? thank you!

tuminoid commented on docs/user-guide/src/capm3/installation_guide.md:
```diff
@@ -11,7 +11,7 @@ install them yourself.
 1. Install `clusterctl`, refer to Cluster API [book](https://cluster-api.sigs.k8s.io/user/quick-start.html#install-clusterctl) for installation instructions.
 1. Install `kustomize`, refer to official instructions [here](https://kubectl.docs.kubernetes.io/installation/kustomize/).
-1. Install Ironic, refer to [this page](../ironic/ironic_installation.html).
+1. Install Ironic, refer to [this page](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/ironic/ironic_installation.md).
```
This link is not broken, why change it? It refers to Ironic installation in the user guide book.

Gutopro replied:
I changed it because when i clicked on the link, it returned a 404 page not found. here's a screenshot of the link destination:
![404](https://github.com/metal3-io/metal3-docs/assets/111025771/d3f726bb-2439-43f2-8299-26f3de211e84)

tuminoid replied:
It is not relative to the repository, but to the user-guide generated site. Use the generated preview site to check if links work or not. The broken links are listed in the issue: https://github.com/metal3-io/metal3-docs/issues/328

Gutopro replied:
thanks for this, it'll help me so much going forward, i'll take it out

tuminoid commented:
> hello @tuminoid I'm in a bit of confusion, I made an initial commit on the branch which I requested a pull from, after some time I fixed another link on the same branch and made a commit on it, now my confusion arose from my inability to create a new pull request, will both commits be tested and treated as a single pull request since they affect the same file and are on the same branch or is there something wrong I did or should do correctly? and also when merging should I use my main branch as the head or should I use the other branch I created? thank you!

Basically, you should create a new branch in your fork for each change, then they can be created as separate PRs. Same branch = same PR. However, since the URLs you're changing are in the same file, you can do it in one PR.

That said, you should squash the commits into one. Generally, the rule is one commit per PR. You also need to address all reviewer's questions. In this case, there is open question on one of the changes.

Gutopro commented:
@tuminoid I just fixed the incorrect changes can you have a look at it and tell me what you think?

tuminoid commented:
> @tuminoid I just fixed the incorrect changes can you have a look at it and tell me what you think?

You have six commits in this PR that do and undo changes, squash them please.

Gutopro commented:
> > @tuminoid I just fixed the incorrect changes can you have a look at it and tell me what you think?
> 
> You have six commits in this PR that do and undo changes, squash them please.

hello @tuminoid i've squashed the commits into one single commit. thank you and good morning

tuminoid commented:
This PR should probably be better pointing at new page https://github.com/metal3-io/metal3-docs/pull/352 is creating/reorganizing.

/hold

Gutopro commented:
> This PR should probably be better pointing at new page #352 is creating/reorganizing.
> 
> /hold

thanks for that @tuminoid  should I wait for his PR to be merged or can i link it directly to his already created page?

Gutopro commented:
> > This PR should probably be better pointing at new page #352 is creating/reorganizing.
> > /hold
> 
> thanks for that @tuminoid should I wait for his PR to be merged or can i link it directly to his already created page?

@tuminoid hello I have updated the link which points to #352 could you kindly review it? thank you

tuminoid commented on docs/user-guide/src/capm3/installation_guide.md:
```diff
@@ -12,7 +12,7 @@ install them yourself.
 1. Install `kustomize`, refer to official instructions [here](https://kubectl.docs.kubernetes.io/installation/kustomize/).
 1. Install Ironic, refer to [this page](../ironic/ironic_installation.html).
-1. Install Baremetal Operator, refer to [TODO](TODO).
+1. Install Baremetal Operator, refer to [this page](https://github.com/metal3-io/metal3-docs/blob/main/docs/user-guide/src/bmo/install_baremetal_operator.md).
```
The link should be relative, just like the ironic installation link is above. `../bmo/install_baremetal_operator.html` or so. We do not point to source files of the book in the book, but to generated content.

You also need to rebase this commit, so the BMO installation guide merged in #352 is visible in your preview as well.

Gutopro replied:
@tuminoid  I have reviewed the link and also rebased the commits could you please review and offer feedback? thank you

tuminoid replied:
You're now including the content from #352 in your commit. You need to rebase, something like this:
- `git fetch --all`
- `get rebase origin/source`
- resolve any  possible conflicts, then `git add <files>`, `git rebase --continue` (skip all if no conflicts)
- git push your branch into your fork

This PR must only contain one line diff where you fix the link.

tuminoid replied:
You can see here https://github.com/metal3-io/metal3-docs/pull/351/files that you're still including more than the link.

Gutopro replied:
yes sir I just figured out that was the problem, I tried the processes you gave me but the files are still there, when i run git rebase origin/main it tells me everything is up to date. it seems the files are embedded in my commit. I'm currently looking for a way to remove those files from my commit

tuminoid replied:
I can't tell you the exact commands to run as I don't know your exact setup, but the idea is to 'git fetch` changes from the this repository (not your fork) and then `git rebase` your fork versus that fetch head. This would remove the parts already in this repository from your PR. If Git tells you `already up to date`, it means you haven't fetched the upstream repo (this repo) or you're rebasing against wrong remote. 

Something like this: https://medium.com/@ruthmpardee/git-fork-workflow-using-rebase-587a144be470

Gutopro replied:
thanks alot @tuminoid the article made it super easy, i really appreciate. it's been fixed.

tuminoid replied:
Looks correct now. :tada:

tuminoid reviewed (commented):
/lgtm
/assign @kashifest 

Thanks for the contribution! May the next one be smoother :)

kashifest reviewed (commented):
/approve

Gutopro commented:
> /lgtm /assign @kashifest
> 
> Thanks for the contribution! May the next one be smoother :)

lol thanks @tuminoid i wouldn't be able to do it without you!

dtantsur commented:
/hold cancel
